src/boot/efi/console.c

```diff
@@ -53,6 +53,8 @@ EFI_STATUS console_key_read(UINT64 *key, BOOLEAN wait) {
                                 return 0;
                         }
                 }
+                else
+                        return err;
```

Comment:
This essentially removes the fallback to ReadKeyStroke below. The patch as such doesn't make sense. It solves the case when ReadKeyStrokeEx works but ReadKeyStroke is buggy, but will break systems where ReadKeyStrokeEx is not available.

I we are calling ReadKeyStroke correctly (and I assume we are), then the only options seems to be to avoid calling ReadKeyStroke on HP. (We print FirmwareVendor in print_status(), so we could do a check against this string here.)

Reply:
From the comment in the source I had seemed to understand that the fallback mode was necessary 
only when the ReadKeyStrokeEx works but doesn't return any key, 
so I thought that the case which 'keypress' <= 0 was enough.

Anyway I will take a look to FirmwareVendor in print_status() and update this PR

Reply:
sadly @medhefgo patch doesn't work, so I took a look to FirmwareVendor and it don't seem a viable solution, the FirmwareVendor string don't contains "HP" but "American Megatrends" so we can't use it to determine if sd-boot is running on an HP machine.

At this point, a configuration option to disable the fallback call can be considered an acceptable idea?